server/service/src/sync/sync_api_v5.rs

```diff
@@ -92,8 +96,24 @@ fn generate_headers(hardware_id: &str) -> HeaderMap {
     headers
 }
 
+async fn check_status(response: Response) -> anyhow::Result<Response> {
+    if !response.status().is_success() {
+        let err = response.text().await?;
+        return Err(anyhow::Error::msg(err));
+    }
+    Ok(response)
+}
+
+async fn to_json<T: DeserializeOwned>(response: Response) -> anyhow::Result<T> {
+    // TODO not owned
+    let response_text = response.text().await?;
```

Comment:
we would only want to do this incase .json fails, method on response owns response though, but can use as_bytes, can do in another PR, added TODO